### DISCUSSION TITLE
Updating the TBC DTB File

### DISCUSSION THREAD

**Discussion by teebr** [OE4T/meta-tegra#1510] [Q&A]
I'm working with an Orin AGX on r35.3.1, and unfortunately I need to have a slightly different device tree for the bootloader versus the OS. With L4T I can do this by setting the `TBCDTB_FILE` variable. But from looking at `flashvars` and `tegra234-flash-helper.sh` it seems like it's hard-coded to use the Linux device tree. I can see this line in the flash helper: ` tbcdtbfilename="$dtb_file"` but then `tbcdtbfilename` seems to  not be used. So my questions are:
- could someone point me to exactly what would need to change in `tegra234-flash-helper.sh` in order to optionally have a different device tree for the bootloader? I'm a bit lost with exactly where `uefi_jetson_with_dtb.bin` gets generated. Should `TBCDTB_FILE` be added to `flashvars` and propagate it through that way, or just modify the flash helper?
- I presume a similar process for adding the device tree to the bootloader binary occurs for generating the BUP/ Capsule? I can't see if/ how the device tree for bootloader feeds into that stuff`

Thanks!

**Comment by madisongh** [comment]
> I can see this line in the flash helper:  tbcdtbfilename="$dtb_file" but then tbcdtbfilename seems to not be used. 

It's used in the signing functions in `odmsign.func`.

> I'm a bit lost with exactly where uefi_jetson_with_dtb.bin gets generated.

That concatenation happens in the python scripts - specifically, `tegraflash_impl_t234.py`, I think. 

> Should TBCDTB_FILE be added to flashvars and propagate it through that way, or just modify the flash helper?

Adding it as a flashvar and updating the helper to use it if it's present (and perhaps defaulting to using the kernel DTB if not) sounds like the right approach.  I think that will cover both the flashing case and the BUP generation case.

**Reply by teebr** [reply]
Thanks Matt, I'll give that a go.
